### PR TITLE
dbaas: Protect against panic if connection details not available.

### PR DESCRIPTION
**Pull request by andrewsomething**
Fixes: https://github.com/digitalocean/terraform-provider-digitalocean/issues/576

### CODE REVIEW

File: digitalocean/datasource_digitalocean_database_cluster.go [hunk 1]
```diff
@@ -176,14 +176,20 @@ func dataSourceDigitalOceanDatabaseClusterRead(ctx context.Context, d *schema.Re
...
+				d.Set("password", db.Connection.Password)
+			}
+
+			if db.PrivateConnection != nil {
+				d.Set("private_host", db.PrivateConnection.Host)
+				d.Set("private_uri", db.PrivateConnection.URI)
+			}
+
 			d.Set("urn", db.URN())
 			d.Set("private_network_uuid", db.PrivateNetworkUUID)
```
Comment (flyte) [inline digitalocean/datasource_digitalocean_database_cluster.go:194]: You'll have to check for the existence of `db.PrivateNetworkUUID` as well, as `private_network_uuid` doesn't exist in the API response when the DB isn't assigned to a VPC at all (any DB created before VPCs were added).

File: digitalocean/resource_digitalocean_database_cluster.go [hunk 2]
```diff
@@ -417,14 +417,20 @@ func resourceDigitalOceanDatabaseClusterRead(ctx context.Context, d *schema.Reso
...
+		d.Set("password", database.Connection.Password)
+	}
+
+	if database.PrivateConnection != nil {
+		d.Set("private_host", database.PrivateConnection.Host)
+		d.Set("private_uri", database.PrivateConnection.URI)
+	}
+
 	d.Set("urn", database.URN())
 	d.Set("private_network_uuid", database.PrivateNetworkUUID)
```
Comment (flyte) [inline digitalocean/resource_digitalocean_database_cluster.go:435]: Same here.

**Review by flyte — CHANGES_REQUESTED** [review]

File: digitalocean/datasource_digitalocean_database_cluster.go (hunk 1 above)
Comment (andrewsomething) [reply digitalocean/datasource_digitalocean_database_cluster.go:194]: The type for a `db.PrivateNetworkUUID` is string. So if it does not exist, it be assigned the default value which is an empty string. The problem that causes the panic is that `db.PrivateConnection` is a struct. The default value of a struct is `nil`. So when you attempt to access an attribute (e.g. `db.PrivateConnection.URI`) it causes a panic as `nil` does not have any attributes.

**Review by ChiefMateStarbuck — APPROVED** [review]
Clean and Mean 👍🏾

File: digitalocean/datasource_digitalocean_database_cluster.go (hunk 1 above)
Comment (flyte) [reply digitalocean/datasource_digitalocean_database_cluster.go:194]: Ahh, OK, thanks for the explanation! I've not used Go yet :)

**Review by flyte — APPROVED** [review]

**Pull request merged**